src/python/dxpy/utils/app_unbuilder.py

```diff
@@ -46,6 +47,21 @@ def _recursive_cleanup(foo):
             if val == "" or val == [] or val == {}:
                 del foo[key]
 
+def flatten_patterns(metadata):
```

Comment:
Can you please add a unit test of this method as well? It would be good to verify what it does on a more or less complete JSON string.